object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -757,29 +769,36 @@ private void checkEnclosingMethodIsCreatesObligation(
     }
 
     Set<String> targetStrsWithoutAdaptation;
+    MustCallAnnotatedTypeFactory mcAtf =
+        typeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
     if (createsObligation != null) {
       targetStrsWithoutAdaptation =
           Collections.singleton(
-              AnnotationUtils.getElementValue(createsObligation, "value", String.class, true));
+              AnnotationUtils.getElementValue(
+                  createsObligation, mcAtf.createsObligationValueElement, String.class, "this"));
     } else {
       // multiple create obligations
       List<AnnotationMirror> createsObligationAnnos =
           AnnotationUtils.getElementValueArray(
-              createsObligations, "value", AnnotationMirror.class, false);
+              createsObligations, mcAtf.createsObligationListValueElement, AnnotationMirror.class);
       targetStrsWithoutAdaptation = new HashSet<>();
       for (AnnotationMirror co : createsObligationAnnos) {
         targetStrsWithoutAdaptation.add(
-            AnnotationUtils.getElementValue(co, "value", String.class, true));
+            AnnotationUtils.getElementValue(
+                co, mcAtf.createsObligationValueElement, String.class, "this"));
       }
     }
-    JavaExpressionContext context =
-        JavaExpressionParseUtil.JavaExpressionContext.buildContextForMethodDeclaration(
-            enclosingMethod, checker);
     String checked = "";
     for (String targetStrWithoutAdaptation : targetStrsWithoutAdaptation) {
-      String targetStr =
-          MustCallTransfer.standardizeAndViewpointAdapt(
-              targetStrWithoutAdaptation, currentPath, context);
+      String targetStr = null;
+      try {
+        targetStr =
+            StringToJavaExpression.atMethodBody(
+                    targetStrWithoutAdaptation, enclosingMethod, checker)
+                .toString();
+      } catch (JavaExpressionParseException e) {
```

Comment:
Here and elsewhere, I'm not sure why the code suppresses an exception.  Doesn't the exception indicate a problem that either the user or the type system designer should know about?

Reply:
We issue custom errors when `@CreatesObligation` annotation arguments are unparseable rather than propagating the exception. I found it more helpful, since the custom error messages can give better context.